File: java/src/org/albanianskills/db/Solver.java
package org.albanianskills.db;

import org.albanianskills.db.models.Database;
import org.albanianskills.db.operations.Operation;
import org.albanianskills.db.operations.OperationParser;

import java.io.*;

public class Solver {

    public static void main(String[] args) throws IOException {

        // initializing the file to read the commands
        File file = new File("hello_world.txt");
        BufferedReader br = new BufferedReader(new FileReader(file));

        // print the results in the output file
        PrintStream fileStream = new PrintStream("output.txt");
        System.setOut(fileStream);

        // get the database instance
        Database db = Database.getInstance();

        // read the input line by line and parse the commands
        String line;
        while ((line = br.readLine()) != null) {
            Operation operation = OperationParser.parseCommand(line, db);
            operation.execute();
            operation.revert();
        }
    }

}

File: nodejs/solver.js
const OperationParser = require('./operations/operation_parser.js')
const Database = require('./models/database.js')


var fs = require('fs');

fs.writeFile('output.txt', '', function(err) {
    if (err) return console.log(err);
});

fs.readFile('input.txt', 'utf8', function(err, data) {
    if (err) throw err;
    operation = OperationParser.parseCommand(data, new Database());
    operation.execute();
    operation.revert();
});